src/main/java/ch/njol/skript/expressions/ExprTarget.java

```diff
@@ -53,80 +55,116 @@
 		setExpr((Expression<? extends LivingEntity>) exprs[1 - matchedPattern]);
 		return true;
 	}
-	
+
 	@Override
-	protected Entity[] get(final Event e, final LivingEntity[] source) {
+	protected Entity[] get(Event e, LivingEntity[] source) {
 		return get(source, new Converter<LivingEntity, Entity>() {
 			@Override
 			@Nullable
-			public Entity convert(final LivingEntity en) {
+			public Entity convert(LivingEntity en) {
 				if (getTime() >= 0 && e instanceof EntityTargetEvent && en.equals(((EntityTargetEvent) e).getEntity()) && !Delay.isDelayed(e)) {
-					final Entity t = ((EntityTargetEvent) e).getTarget();
-					if (t == null || type != null && !type.isInstance(t))
+					Entity target = ((EntityTargetEvent) e).getTarget();
+					if (target == null || type != null && !type.isInstance(target))
 						return null;
-					return t;
+					return target;
 				}
-				return Utils.getTarget(en, type);
+				return getTarget(en, type);
 			}
 		});
 	}
-	
+
 	@Override
 	public Class<? extends Entity> getReturnType() {
 		return type != null ? type.getType() : Entity.class;
 	}
-	
+
 	@Override
-	public String toString(final @Nullable Event e, final boolean debug) {
+	public String toString(@Nullable Event e, boolean debug) {
 		if (e == null)
 			return "the target" + (type == null ? "" : "ed " + type) + (getExpr().isDefault() ? "" : " of " + getExpr().toString(e, debug));
 		return Classes.getDebugMessage(getAll(e));
 	}
-	
+
 	@Override
-	public boolean setTime(final int time) {
+	public boolean setTime(int time) {
 		return super.setTime(time, EntityTargetEvent.class, getExpr());
 	}
-	
+
 	@Override
 	@Nullable
-	public Class<?>[] acceptChange(final ChangeMode mode) {
+	public Class<?>[] acceptChange(ChangeMode mode) {
 		if (mode == ChangeMode.SET || mode == ChangeMode.DELETE)
 			return CollectionUtils.array(LivingEntity.class);
 		return super.acceptChange(mode);
 	}
-	
+
 	@Override
-	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
 		if (mode == ChangeMode.SET || mode == ChangeMode.DELETE) {
-			final LivingEntity target = delta == null ? null : (LivingEntity) delta[0];
-			for (final LivingEntity entity : getExpr().getArray(e)) {
+			LivingEntity target = delta == null ? null : (LivingEntity) delta[0];
+			for (LivingEntity entity : getExpr().getArray(e)) {
 				if (getTime() >= 0 && e instanceof EntityTargetEvent && entity.equals(((EntityTargetEvent) e).getEntity()) && !Delay.isDelayed(e)) {
 					((EntityTargetEvent) e).setTarget(target);
-				} else {
-					if (entity instanceof Creature)
-						((Creature) entity).setTarget(target);
+				} else if (entity instanceof Mob) {
+					((Mob) entity).setTarget(target);
 				}
 			}
-		} else {
-			super.change(e, delta, mode);
+			return;
 		}
+		super.change(e, delta, mode);
```

Comment:
I don't think we need to do this - it may make unintended changes to the entity

Reply:
Didn't change this, GitHub just highlighted it due to formatting. This is existing behavior.

Reply:
Having this outside the else statement will call the DefaultChanger.entityChanger for the cases SET and DELETE, this shouldn't be done as these have already been handled above.

Reply:
It has a return, not an else statement. Meaning the super.acceptChange will check for any case that's not SET or DELETE.

Reply:
Didn't see the return, in that case it's fine